packages/synapse-sdk/src/storage/context.ts

```diff
@@ -175,6 +178,110 @@ export class StorageContext {
     this._pdpServer = new PDPServer(authHelper, provider.products.PDP.data.serviceURL)
   }
 
+  /**
+   * Creates new storage contexts with specified options
+   * Each context correseponds to a different data set
+   */
+  static async createContexts(
+    synapse: Synapse,
+    warmStorageService: WarmStorageService,
+    options: CreateContextsOptions = {}
+  ): Promise<StorageContext[]> {
+    const count = options?.count ?? 2
+    const resolutions: ProviderSelectionResult[] = []
+    const clientAddress = await synapse.getClient().getAddress()
+    const registryAddress = warmStorageService.getServiceProviderRegistryAddress()
+    const spRegistry = new SPRegistryService(synapse.getProvider(), registryAddress)
+    if (options?.dataSetIds) {
+      for (const dataSetId of new Set(options.dataSetIds)) {
+        const resolution = await StorageContext.resolveByDataSetId(
```

Comment:
`Promise.all()` would be good here to do this in parallel. But you have the resolutions.length==count thing going on, so you could then slice off `count` (max) ids from `Set(options.dataSetId)` and only do those, then you're going to at least be under `count`.

I'd look for options to do the same with the others, maybe you could even just accrue `promises` into an array for all of the various resolution steps here, skip the `await` on each of them so there's no awaiting until the very end and keep `promises.length <= count` and then `await Promise.all(promises)`